packages/angular/src/executors/browser-esbuild/browser-esbuild.impl.ts

```diff
@@ -0,0 +1,82 @@
+import { type EsBuildSchema } from './schema';
+import { validateOptions } from './lib/validate-options';
+import { type DependentBuildableProjectNode } from '@nx/js/src/utils/buildable-libs-utils';
+import { type ExecutorContext, readCachedProjectGraph } from '@nx/devkit';
+import { createTmpTsConfigForBuildableLibs } from './lib/buildable-libs';
+import { createBuilderContext } from 'nx/src/adapter/ngcli-adapter';
+import { type BuilderOutput } from '@angular-devkit/architect';
+import { type OutputFile } from 'esbuild';
+import { createAsyncIterable } from '@nx/devkit/src/utils/async-iterable';
+
+export default async function* esbuildExecutor(
+  options: EsBuildSchema,
+  context: ExecutorContext
+) {
+  validateOptions(options);
+  options.buildLibsFromSource ??= true;
+
+  const { buildLibsFromSource, ...delegateExecutorOptions } = options;
+
+  let dependencies: DependentBuildableProjectNode[];
+  let projectGraph = context.projectGraph;
+
+  if (!buildLibsFromSource) {
+    projectGraph = projectGraph ?? readCachedProjectGraph();
+    const { tsConfigPath, dependencies: foundDependencies } =
+      createTmpTsConfigForBuildableLibs(
+        delegateExecutorOptions.tsConfig,
+        context,
+        { projectGraph }
+      );
+    dependencies = foundDependencies;
+    delegateExecutorOptions.tsConfig = tsConfigPath;
+  }
+  const { buildEsbuildBrowser } = await import(
+    '@angular-devkit/build-angular/src/builders/browser-esbuild/index'
+  );
+
+  const builderContext = await createBuilderContext(
+    {
+      builderName: 'browser-esbuild',
+      description: 'Build a browser application',
+      optionSchema: await import(
+        '@angular-devkit/build-angular/src/builders/browser-esbuild/schema.json'
+      ),
+    },
+    context
+  );
+
+  const builder = buildEsbuildBrowser(
+    delegateExecutorOptions,
+    builderContext
+  ) as AsyncIterable<
+    BuilderOutput & {
+      outputFiles?: OutputFile[];
+      assetFiles?: { source: string; destination: string }[];
+    }
+  >;
+
+  async function* runBuilder() {
+    return yield* builder;
+  }
+
+  return yield* createAsyncIterable(async ({ next, done }) => {
+    const iter = runBuilder();
+    let buildRes = await iter.next();
+    while (true) {
+      if (buildRes.done === true) {
+        if (buildRes.value) {
+          next({ success: buildRes.value.success });
+        }
+        break;
+      } else if (!buildRes.value.success) {
+        next({ success: buildRes.value.success });
+        break;
+      } else {
+        next({ success: buildRes.value.success });
+      }
+      buildRes = await iter.next();
+    }
+    done();
+  });
```

Comment:
Tried this again and this is needed because of how Angular creates the Async Iterable. We need to loop through until it completes. The same pattern is used in other packages such as the Module Federation Dev Server in the React Package